examples/efficientdet/efficientdet_building_blocks.py

```diff
@@ -0,0 +1,321 @@
+        if height > target_height and width > target_width:
+            if not self.conv_after_downsample:
+                feat = self._maybe_apply_1x1(feat, training, num_channels)
+            feat = self._pool2d(feat,
+                                height,
+                                width,
+                                target_height,
+                                target_width)
+            if self.conv_after_downsample:
+                feat = self._maybe_apply_1x1(feat, training, num_channels)
+        elif height <= target_height and width <= target_width:
+            feat = self._maybe_apply_1x1(feat, training, num_channels)
+            if height < target_height or width < target_width:
+                feat = self._upsample2d(feat, target_height, target_width)
+        else:
+            raise ValueError('Incompatible Resampling : feat shape {}x{} \
+            target_shape: {}x{}'.format(height,
+                                        width,
+                                        target_height,
+                                        target_width))
+
+        return feat
+
+
+class FPNCells(tf.keras.layers.Layer):
+    """Set of FPN Cells."""
+
+    def __init__(self, config, name='fpn_cells'):
+        super().__init__(name=name)
+        self.config = config
+        self.fpn_config = get_fpn_configuration(config['fpn_name'],
+                                                config['min_level'],
+                                                config['max_level'],
+                                                config['fpn_weight_method']
+                                                )
+        self.cells = [FPNCell(self.config, name='cell_%d' % repeats)
+                      for repeats in range(self.config['fpn_cell_repeats'])]
+
+    def call(self, feats, training):
+        for cell in self.cells:
+            cell_feats = cell(feats, training)
+            min_level = self.config['min_level']
+            max_level = self.config['max_level']
+
+            feats = []
+            for level in range(min_level, max_level + 1):
+                for i, fnode in enumerate(reversed(self.fpn_config['nodes'])):
+                    if fnode['feat_level'] == level:
+                        feats.append(cell_feats[-1 - i])
+                        break
+        return feats
+
+
+class FPNCell(tf.keras.layers.Layer):
+    """A single FPN cell."""
+    def __init__(self, config, name='fpn_cell'):
+        super().__init__(name=name)
+        self.config = config
+        self.fpn_config = get_fpn_configuration(config['fpn_name'],
+                                                config['min_level'],
+                                                config['max_level'],
+                                                config['fpn_weight_method']
+                                                )
+        self.fnodes = []
+        for i, fnode_cfg in enumerate(self.fpn_config['nodes']):
```

Comment:
maybe we can use `node_arg` instead of i or

Reply:
Done